design/one-pager-host-aware-stack-manager.md

```diff
@@ -0,0 +1,211 @@
+- Option 1: Using `in cluster config` for Host Cluster: Host Cluster already auto-mounts service account tokens which we
+  can use.
+
+- Option 2: Using `in cluster config` for Tenant Instance: This will be consistent with other controllers (Crossplane
+  and stack controllers).
+
+To leverage existing mechanisms to authenticate to host cluster, we will use Option 1. We will define RBAC rules for
+stack manager which will be part of `crossplane-controllers` helm chart (see [packaging changes](#packaging-changes))
+that will be deployed into host cluster. This way, we can explicitly declare permissions of stack manager on Host
+Cluster.
+
+#### Stack Installation
+
+In case of a stack installation, Stack Manager should:
+
+1. Create install-job on Host Cluster and read logs of install pod from Host.
+2. Deploy `Stack`, `CRDs` and `RBAC` resources on Tenant Instance
+3. Wait until a service account token secret is created for the stack to be installed.
+4. Copy service account token secret into Host Cluster.
+5. Deploy stack controller deployment by [configuring against Tenant Instance API Server](#configuring-a-pod-against-tenant-kubernetes-api-server).
+
+
+##### Mapping Resources to Host
+
+While creating resources (token secrets or controller deployments) on Host Cluster, Stack Manager will map objects
+(jobs, deployments, secrets ...) on multiple namespaces of Tenant Instance into a single namespace on Host Cluster
+by prepending `<namespace>.` to object name. For example:
+
+ | Tenant Instance | Host Cluster |
+ | :---: | :---: |
+ | Namespace / Name | Namespace / Name |
+ | crossplane-system / stack-gcp | tenant-n-controllers / crossplane-system.stack-gcp |
+ | dev / stack-wordpress | tenant-n-controllers / dev.stack-wordpress |
+
+This mapping will cause a length limit to resource names as follows:
+
+```
+len(namespace) + len(resource-name) + 1 <= 253
+```
+
+Right now, this will effect:
+- Install job name
+- Stack controller deployment name
+- Stack controller service account name
+
+However, considering the following, this should not cause a new limitation for Crossplane:
+
+- There are already labels holding some resource names (e.g. `core.crossplane.io/parent-name: stack-gcp`) and Kubernetes
+ Labels are [limited to 63 chars](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set)
+- Kubernetes namespaces are limited to 63 chars
+ 
+#### Stack Uninstallation
+
+Stack manager can no longer rely on owner references for deletion of stack install jobs and controller deployments
+since stack related custom resources and jobs/deployments live in different Kubernetes API Servers. Label based cleanup
+will be implemented as a solution.
+
+One extra artifact that needs to deleted per Stack uninstallation is the token secret created on Host Cluster. This will
+be achieved by setting owner of that secret as Stack Controller Deployment which also lives same Cluster/Namespace.
+
+### Packaging Changes
```

Comment:
I keep getting confused by this section title each time i read it, I keep thinking it is proposing changes to the Stack package format.  Can you update the section title to something like "Crossplane Installation Changes" or "Crossplane Helm Chart Changes"? 🙇

Reply:
Sure, done :)